### DISCUSSION TITLE
Passing Vectors arguments to the CLI

### DISCUSSION THREAD

**Discussion by yahya010** [daniele77/cli#119] [Q&A] [answered]
Hello,
I am trying to pass a vector as the argument into a function that is called by CLI, but I am getting this error. Do you know how I can do that?
Function:

    static string ReadReg = "ReadReg";
    auto callReadReg = [](std::ostream& out, const std::vector <uint32_t> &regAddr, double wv) {
        for(int i=0; i < regAddr.size(); i++){
                std::cout << regAddr.at(i) << ' ';
        }

Error:
    
    error: ‘Name’ is not a member of ‘cli::TypeDesc<std::vector<unsigned int> >’
      760 |             out << " " << TypeDesc< typename std::decay<P>::type >::Name();

**Comment by daniele77** [comment, marked as answer]
@yahya010 currently `cli` only supports basic types and a vector of strings.
So, you can't use `std::vector<uint32_t>`. 
You have two alternatives:

- if you have a fixed number of `uint32_t `parameters, you can explicit each argument
- if you have a variable number of `uint32_t`, you can use a`std::vector<std::string>` and in your handle, you must convert each string into a uint32_t using e.g., `stoul`

However, soon cli will be able to manage user defined types. See this issue #69
